code/Protobuf.md

```diff
@@ -4,32 +4,36 @@ This document outlines the steps for creating or modifying a protobuf.
 
 ## Requirements
 
-- Install GRPC.
+- Install the protoc compiler that is used to generate protobuf code and gRPC service code, `protoc`, by downloading the binary https://github.com/protocolbuffers/protobuf/releases.  Then move `protoc` to `/usr/local/bin/` and also move `protoc/include` to `/usr/local/include/`. Alternately, you can install protoc using the appropriate package for your OS:
 
-`$ go get -u google.golang.org/grpc`
+  - Debian/Ubuntu/derivatives:
 
-- Install the protoc compiler that is used to generate gRPC service code, `protoc` by downloading the binary https://github.com/protocolbuffers/protobuf/releases.  Then move `protoc` to `/usr/local/bin/` and also move `protoc/include` to `/usr/local/include/`.
+      apt install protobuf-compiler
 
-- Install protoc plugin for Go, [`proto-gen-go`](https://github.com/golang/protobuf#installation):
+  - MacOS with Homebrew:
 
-`$ go get -u github.com/golang/protobuf/protoc-gen-go`
+      brew install protobuf
 
-- Install [`protolock`](https://github.com/nilslice/protolock):
+- Install a protobuf specification linter, the protolock tool, and their dependencies. (This step is idempotent, so run it again if you're not sure whether your tools need updating):
 
-`$ go get -u github.com/nilslice/protolock/...`
+    $ go run scripts/protobuf.go install
 
 ## Steps to Modify `.proto` Files
 
 Assumes all steps occur from the home directory of the https://github.com/storj/storj repo.
 
 1. Modify the `.proto` file
 
-2. From the `pkg/pb/` directory run the following command:
+2. To update the generated code related to your `.proto` file changes, use `go generate` on the pb directory:
 
-`$ go generate`
+    $ go generate ./pkg/pb/...
 
-3. If a check that fails when there's a breaking API change, this command may need to be executed to override:
+3. Update the proto.lock file to match your changes:
 
-`$ protolock commit --force`
+    $ protolock commit
 
-** Note: overriding the breaking API is only allowed during alpha/beta development.
+4. If step 3 fails with an error, your changes are not backward compatible! You should fix that, because we need to maintain backward compatibility so that versions of storagenode/satellite/uplink/etc built at different times can still communicate with each other. We should not break backward compatibility without _very_ careful investigation of the ramifications. If we are sure that the change is still ok, you can use the `--force` argument to protolock to override its complaint:
```

Comment:
What about a feature that hasn't being deployed but parts of it are in master? That should be ok right?

Reply:
If none of the protobufs are being used by anything yet (so, maybe the protobufs exist and there is code that uses them, but that code is not called in master), then that should be ok. Do you think that's worth mentioning in the doc?

Reply:
Maybe it can be one of the ok scenarios listed in the doc?